pym/bob/input.py

```diff
@@ -947,6 +952,12 @@ def getScmDirectories(self):
             dirs.update(s.getDirectories())
         return dirs
 
+    def getCheckoutAsserts(self):
```

Comment:
That should not be needed. What **is** needed is that `getDigestScript` also takes the assertions into account. I guess you introduced `getCheckoutAssert` and treated it like an SCM to trigger the checkout step if the asserts are changed. But this should be really be done through `getDigestScript`. Otherwise release mode builds will not be affected.